### ISSUE TITLE
Unable to find schema when $id starts with #

### ISSUE THREAD

**Issue by michaelmior** [corvus-dotnet/Corvus.JsonSchema#473]
The following schema fails to compile

```json
{
  "$id": "#Configuration",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "properties": {
    "language": {"type": "string"}
  }
}
```

```
System.InvalidOperationException: Unable to find the schema at https://endjin.com/home/mmior/apps/jsonschema-benchmark/implementations/corvus/schema.json#/properties/language
```

**Comment by michaelmior** [comment]
On a related note, is there any particular reason why the URI doesn't just start with `file://` instead of an unrelated domain name?

**Comment by mwadams** [comment]
I didn't think fragment IDs were permitted?

**Comment by michaelmior** [comment]
@mwadams You might be right. I raised this because it's based on a real schema that I found and it's accepted by other validators.

**Comment by mwadams** [comment]
"This keyword must be set to an absolute URI or a relative reference as defined by RFC 3986 without a fragment"

**Comment by mwadams** [comment]
(so it shouldn't be accepted as a valid schema)

**Comment by mwadams** [comment]
Although that qualification is not explicitly present in draft7... Given applying a fragment cannot produce a base URI I'm not sure how that is supposed to work, though!

**Comment by mwadams** [comment]
> On a related note, is there any particular reason why the URI doesn't just start with `file://` instead of an unrelated domain name?

We avoid producing URIs based on the local file system wherever possible. We had all sorts of issues in the past when people were working across UNIX/Windows/MacOS and just decided to avoid the whole can of worms in this case.

**Comment by michaelmior** [comment]
> We avoid producing URIs based on the local file system wherever possible. We had all sorts of issues in the past when people were working across UNIX/Windows/MacOS and just decided to avoid the whole can of worms in this case.

Got it. In that case, I think using just `localhost` for the hostname would be more intuitive, but I guess it doesn't make a difference.

**Comment by michaelmior** [comment]
@mwadams FWIW, I updated the schema in our test suite, so when the Corvus implementation is eventually merged, the case in question should pass.

**Comment by mwadams** [comment]
I'm going to investigate further - I think this may be an issue specifically at the root.

**Comment by mwadams** [comment]
Fixed in 4.0.0-preview.30

Thanks!
